### PR TITLE
Replace eliot_logged_test with something ... else

### PR DESCRIPTION
**Pull request by exarkun**
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3600

### CODE REVIEW

File: src/allmydata/test/eliotutil.py [hunk 1]
```diff
@@ -174,10 +99,83 @@ def addCleanup(self, *a, **kw):
...
+            setattr(self.case, name, with_logging(ensure_text(self.case.id()), original))
+            # Then use the rest of the machinery to run it.
+            return self._run_tests_with_factory(
+                self.case,
+                self.handlers,
+                self.last_resort,
+            ).run(result)
+        finally:
+            # Clean up the patching for idempotency or something.
+            delattr(self.case, name)
```
Comment (meejah) [inline src/allmydata/test/eliotutil.py:133]: For proper cleanup, `original` should be set again I think.

File: src/allmydata/test/eliotutil.py [hunk 2]
```diff
@@ -174,10 +99,83 @@ def addCleanup(self, *a, **kw):
...
+                    check_for_errors(validating_logger)
+        finally:
+            swap_logger(original)
+    return run_with_logging
+
+
+class _TwoLoggers(object):
+    """
+    Log to two loggers.
+
```
Comment (meejah) [inline src/allmydata/test/eliotutil.py:169]: So "a logger" is just anything that has a `.write` method?

**Review by meejah — APPROVED** [review]
Looks good. One inline thing to fix then merge.

File: src/allmydata/test/eliotutil.py (hunk 2 above)
Comment (exarkun) [reply src/allmydata/test/eliotutil.py:176]: Almost.  Added more docs to make this clear.

**Pull request merged**